rpc/handlers.go

```diff
@@ -734,3 +738,26 @@ func setEventFilterRange(filter *blockchain.EventFilter, fromID, toID *BlockID,
 	}
 	return set(blockchain.EventFilterTo, toID)
 }
+
+// AddInvokeTransaction relays an invoke transaction to the gateway.
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_write_api.json#L11
+func (h *Handler) AddInvokeTransaction(invokeTx *gateway.BroadcastedInvokeTxn) (*AddInvokeTransactionResponse, *jsonrpc.Error) {
+	if invokeTx.Type == "INVOKE" {
+		invokeTx.Type = "INVOKE_FUNCTION"
+	}
```

Comment:
This shouldnt be necessary, right? It is typed `BroadcastedInvokeTxn` anyways.

Reply:
BroadcastedInvokeTxn is part of the rpc spec. so, IMO, it should be in `rpc` package.

Reply:
The thing about this hack is that rpc spec says `Type` is `INVOKE` and pathfinder follows the spec.
But gateway expects type to be `INVOKE_FUNCTION`. It would be great if we could just pass recevied `BroadcastedInvokeTxn` to gateway but unfortunately things get complicated here.

Regarding moving it to `rpc` package sounds reasonable.

Reply:
As discussed on daily last Friday: now it's going to be always INVOKE_FUNCTION

Reply:
Regarding moving to `rpc` package: it causes import cycle, which could be solved by introducing private gateway interface in `rpc` package itself:
```
//go:generate mockgen -destination=../../mocks/mock_gateway.go -package=mocks github.com/NethermindEth/juno/clients/gateway Gateway
type gateway interface {
	AddInvokeTransaction(context.Context, *BroadcastedInvokeTxn) (*InvokeResponse, error)
}
```
but in this case type `gateway.InvokeResponse` introduces another cycle :). `InvokeResponse` is type related to gateway communication and shouldn't be moved to `rpc` package. So I think we should keep it as is, we can solve it by copying same type in rpc and gateway packages but it's overkill IMO

Reply:
`InvokeResponse` can stay in `gateway`. But structs like `BroadcastedTxn`, `InvokeTxnV1` & `BroadcastedInvokeTxn` should be part of the `rpc` package

for example; 

[getEstimateFee](https://github.com/starkware-libs/starknet-specs/blob/aaea417f193bbec87b59455128d4b09a06876c28/api/starknet_api_openrpc.json#L487-L504) has nothing to do with gateway, yet it still uses broadcasted tx types.

Gateway shouldnt depend on and RPC types. Duplicating the types to achieve this is not that big of an issue, we have duplicated types between `feeder` and `core` anyways.

Reply:
After recent discussions, structs related to addInvokeTx (eg `BroadcastedTxn`) will be moved to a new package `sequencer`, which `rpc` and `gateway` will import. This means rpc and gateway don't have to import each other.